### ISSUE TITLE
Updated results for distribution drift issue

### ISSUE THREAD

**Issue by jyaacoub** [jyaacoub/MutDTA#131]
There is a significant distribution drift due to the new training split we had created to exclude any proteins in OncoKB...
- This is evident by the **drastic difference in performance when training with the split vs when training on a random split.**
- Most evident in the PDBbind dataset.
- [ ] get distributional stats to provide additional support for this claim
   - Via sequence identity?

## TODOs:
- [x] build new dataset with a random, but unified test set (any overlaping proteins across datasets that are in the test set, must also be in the test set).
- [x] Identify the list of ligand-protein pairs from oncoKB that are not in the training set, which can be used for Mutagenesis analysis.
- [ ] profit??

**Comment by jyaacoub** [comment]
# Get test OncoKBs
## PDBbind 
Using both uniprot ID and pdb ID:
![image](https://github.com/user-attachments/assets/6ec24758-a3bd-4f52-b313-382142691ebb)
Filtering by both gene name and drug name is difficult since PDBbind uses shortform names like "NLG" and the oncoKB uses the full name of the drug, and there are multiple aliases for each drug so the shortform name might not match. We would have to check against all possible aliases to ensure that we had the right drug-target pair.
<details><summary>code</summary>
<p>

```python
#%%
import pandas as pd

def get_test_oncokbs(train_df=pd.read_csv('/cluster/home/t122995uhn/projects/data/test/PDBbindDataset/nomsa_binary_original_binary/full/cleaned_XY.csv'),
                     oncokb_fp='/cluster/home/t122995uhn/projects/data/tcga/mart_export.tsv', 
                     biomart='/cluster/home/t122995uhn/projects/downloads/oncoKB_DrugGenePairList.csv'):
    #Get gene names for PDBbind
    dfbm = pd.read_csv(oncokb_fp, sep='\t')
    dfbm['PDB ID'] = dfbm['PDB ID'].str.lower()
    train_df.reset_index(names='idx',inplace=True)

    df_uni = train_df.merge(dfbm, how='inner', left_on='prot_id', right_on='UniProtKB/Swiss-Prot ID')
    df_pdb = train_df.merge(dfbm, how='inner', left_on='code', right_on='PDB ID')

    # identifying ovelap with oncokb
    # df_all will have duplicate entries for entries with multiple gene names...
    df_all = pd.concat([df_uni, df_pdb]).drop_duplicates(['idx', 'Gene name'])[['idx', 'code', 'Gene name']]

    dfkb = pd.read_csv(biomart)
    df_all_kb = df_all.merge(dfkb.drop_duplicates('gene'), left_on='Gene name', right_on='gene', how='inner')

    trained_genes = set(df_all_kb.gene)

    #Identify non-trained genes
    return dfkb[~dfkb['gene'].isin(trained_genes)]


train_df = pd.read_csv('/cluster/home/t122995uhn/projects/data/test/PDBbindDataset/nomsa_binary_original_binary/train0/cleaned_XY.csv')
val_df = pd.read_csv('/cluster/home/t122995uhn/projects/data/test/PDBbindDataset/nomsa_binary_original_binary/val0/cleaned_XY.csv')

train_df = pd.concat([train_df, val_df])

get_test_oncokbs(train_df=train_df)
``` 

</p>
</details> 

## Davis

**Comment by jyaacoub** [comment]
# Still need to get updated results!

**Comment by jyaacoub** [comment]
# stats on distribution differences:
Small difference in the % of heavily targeted proteins that appear in the test set (oncoKB test set had fewer proteins that were heavily targeted)
- Likely due to the fact that we had hand picked oncoKB proteins which were heavily targeted and so the rest of the proteins were low targets, and thus the majority ended up being low targets.

## Distribution test splits:
#### Full distribution:
We can see that the oncoKB test split has way more "lowly" targeted proteins in the test set, which are naturally going to be harder for the model to be evaluated on due to the range of proteins that the model has to be good at.
The reason for this was because we hand picked the more heavily targeted proteins for the test set, and because we want the test sets to remain the same size, the code that I wrote picked lowly targeted proteins in order to reach that size constraint.
Whereas the random split has a more even distribution, mimicing more closely the training split.
![image](https://github.com/user-attachments/assets/63b27643-82e6-40c0-af8f-58727ad0f84c)

#### Limit to top n frequent proteins
This really shows the impact of hand picking the proteins for the test set. They are shown in the spikes of green from the oncokb test split.
![image](https://github.com/user-attachments/assets/0b1c4509-a11b-48bb-9f98-ab729f3ffc8f)
![image](https://github.com/user-attachments/assets/d53d5d00-fdee-45cd-b2a0-c6a6f5bacd3e)
![image](https://github.com/user-attachments/assets/3b065e27-0949-463c-9978-0b702dd49433)
